packages/teleterm/src/services/tshdEvents/createTshdEventsServer.ts

```diff
@@ -0,0 +1,41 @@
+import { Server, ServerCredentials } from '@grpc/grpc-js';
+
+import Logger from 'teleterm/logger';
+
+/**
+ * Starts tshd events server.
+ * @return {Promise} Object containing an instance of the server and the address it's listening on.
+ */
+export async function createTshdEventsServer(
+  requestedAddress: string,
+  credentials: ServerCredentials
+): Promise<{ server: Server; resolvedAddress: string }> {
+  const logger = new Logger('tshd events');
+  const server = new Server();
+
+  // grpc-js requires us to pass localhost:port for TCP connections,
+  const grpcServerAddress = requestedAddress.replace('tcp://', '');
+
+  return new Promise((resolve, reject) => {
+    try {
+      server.bindAsync(grpcServerAddress, credentials, (error, port) => {
+        if (error) {
+          reject(error);
+          return logger.error(error.message);
```

Comment:
As far as I can tell, we return here just to stop the execution of the rest of the code so I don't think we can return then reject.

But I'll update this so that we return undefined rather than suggesting that we need to return whatever the return of `logger.error` is. It's probably undefined too but I think it'd be better to have it on a separate line.